extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,304 @@
+    containerElement.setAttribute(
+      'id', '-amp-font-container-' + ncustomFontFamily_);
+    style.setStyles(containerElement, {
+      fontSize: '48px',
+      fontVariant: 'normal',
+			left: '-999px',
+      lineHeight: 'normal',
+      margin: 0,
+      padding: 0,
+      position: 'absolute',
+			top: '-999px'
+    });
+    DEFAULT_FONTS_.forEach(font => {
+      let defaultFontElement = document.createElement('div');
+      let nfont = normalizeFontFamily_(font);
+      let defaultFontElementId = '-amp-font-'+ ncustomFontFamily_ + '-' + nfont;
+      defaultFontElement.setAttribute('id', defaultFontElementId);
+      defaultFontElement.textContent = TEST_STRING_;
+      style.setStyles(defaultFontElement, {
+        fontFamily: font,
+        margin: 0,
+        padding: 0,
+        whiteSpace: 'nowrap'
+      });
+      containerElement.appendChild(defaultFontElement);
+    });
+    document.body.appendChild(containerElement);
+    DEFAULT_FONTS_.forEach(font => {
+      let nfont = normalizeFontFamily_(font);
+      let defaultFontElementId = '-amp-font-'+ ncustomFontFamily_ + '-' + nfont;
+      let element = document.getElementById(defaultFontElementId);
+      this.measurements_[defaultFontElementId] = {
+        width: element./*OK*/offsetWidth,
+        height: element./*OK*/offsetHeight
+      };
+    });
+  }
+
+
+  /**
+   * Make the browser that don't support font loading events to download the
+   * custom font by creating an element (with text) not visible on the viewport.
+   * font download is detected by comparing the elements height and width with
+   * measurements taken in step1.
+   * @private
+   */
+  loadFont_() {
+    var haswidthChanged = false;
+    var hasHeightChanged = false;
+    var ncustomFontFamily_ = normalizeFontFamily_(this.customFontFamily_);
+    DEFAULT_FONTS_.some(font => {
+      if(this.hasTimedOut_) {
+        this.onFontLoadError_();
+        return true;
+      }
+      let nfont = normalizeFontFamily_(font);
+      let defaultFontElementId = '-amp-font-'+ ncustomFontFamily_ + '-' + nfont;
+      let element = document.getElementById(defaultFontElementId);
+      let currentFontFamily = element.style.fontFamily;
+      if (currentFontFamily.indexOf(this.customFontFamily_) == -1) {
+        element.style.fontFamily =
+            this.customFontFamily_ + ',' + currentFontFamily;
+      }
+      var defaultDimensions =
+          this.measurements_[defaultFontElementId];
+      haswidthChanged = (
+        Math.abs(
+          defaultDimensions.width - element./*OK*/offsetWidth) > TOLERANCE_);
+      hasHeightChanged = (
+        Math.abs(
+          defaultDimensions.height - element./*OK*/offsetHeight) > TOLERANCE_);
+      if (haswidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if (haswidthChanged || hasHeightChanged) {
+      this.onFontLoadSuccess_();
+    } else if(this.hasTimedOut_) {
+      this.onFontLoadError_();
+      return true;
```

Comment:
this is sorta weird, so this function can return `boolean` or not return anything?